### ISSUE TITLE
Default weapon selection suboptimal

### ISSUE THREAD

**Issue by vgaming** [wesnoth/wesnoth#2337]
This weapon gets sometimes auto-selected when trying to attack an enemy.

![weapon-selection](https://user-images.githubusercontent.com/33452702/34421278-61ee0164-ec1f-11e7-9d63-ea33714c5656.png)

I understand the difficulties of solving this problem, yet I still think this can be counted as a bug. Thoughts?

Possible related to https://github.com/wesnoth/wesnoth/issues/1807  as this ticket discusses "provably better" comparisons, too.

I'm OK with closing the issue if we acknowledge the problem and acknowledge that there won't be any solution. (So not to leave stale issues.)

**Comment by Wedge009** [comment]
I'm just wondering if this is really a bug since the game isn't really required to suggest an 'optimal' weapon for the player, is it? Otherwise what's the point of having a weapon choice?

It's a good pick-up, though, and the link to #1807 is a good find as well.

**Comment by vgaming** [comment]
@Wedge009 well the point of chosing weapon might be to select a desired taken damage values and/or special effects (slow, poison).
I agree with you that "always better" comparisons are not documented anywhere, so you can't really say that it's a 100% bug.
I guess the real reason why this ticket exists is:
* fix the underlying problem of AI battle estimation, where slow is not honored by itself. But should, for better AI play if nothing else
* make better weapon selection to reduce the unexpected behavior for normal users. Obviously, people get used to the fact that auto-selected weapon is good. Look at this with gamer's eyes: you really _are_ surprised why a better weapon was not chosen, when it's so obvious that it's better.

**Comment by Wedge009** [comment]
I understand your points, but personally I _don't_ expect the game to pick a weapon for me. I'm the player, not the game! Honestly, I've never really noticed, but I thought the top option is always selected (if there's more than one).

If we want to have the 'best' option selected by default (if it is indeed possible to determine an absolute best option), then I don't mind that we do so if it's a deliberate design decision. I just didn't think or expect that Wesnoth was required to do this.

**Comment by CelticMinstrel** [comment]
Given that the "best" weapon selected here is (if I recall correctly) also the "best" weapon that would be selected by the AI in this circumstance, this could technically be treated as an AI bug,

**Comment by cooljeanius** [comment]
> Given that the "best" weapon selected here is (if I recall correctly) also the "best" weapon that would be selected by the AI in this circumstance, this could technically be treated as an AI bug,

Yeah, I have always thought it to be weird how the AI doesn't know about "slow"... (although the recent switch to the experimental AI as the new default in 1.16 seems to have taught the AI about poison, at least... maybe the new `spread_poison` CA could be duplicated and repurposed into a new `spread_slowness` CA?)
